src/data/list/basic.lean

```diff
@@ -1149,6 +1149,26 @@ theorem index_of_lt_length {a} {l : list α} : index_of a l < length l ↔ a ∈
 ⟨λh, decidable.by_contradiction $ λ al, ne_of_lt h $ index_of_eq_length.2 al,
 λal, lt_of_le_of_ne index_of_le_length $ λ h, index_of_eq_length.1 h al⟩
 
+theorem index_of_append_of_mem {a : α} (h : a ∈ l₁) :
+  index_of a (l₁ ++ l₂) = index_of a l₁ :=
+begin
+  induction l₁,
+  { exfalso, exact not_mem_nil a h },
+  rw list.cons_append,
+  by_cases hh : a = l₁_hd,
+  { iterate 2 { rw index_of_cons_eq _ hh } },
+  rw [index_of_cons_ne _ hh, index_of_cons_ne _ hh, l₁_ih (mem_of_ne_of_mem hh h)],
```

Comment:
better specify the names when doing `induction` rather than using auto-generated `l₁_hd` `l₁_ih`.